tests/integration/molecule/module_secrets_provider_vault/converge.yml

```diff
@@ -0,0 +1,277 @@
+---
+- name: Pre-converge step - ensure presence of valid PEM files for TLS configuration
+  hosts: all
+  gather_facts: no
+  tasks:
+    # As part of configuring TLS for the vault secrets provider, we need
+    # to specify paths to CA cert and client cert/key files.
+    # If these files do not exist at the specified paths on SensuGo backend,
+    # or are not valid PEM files, the API returns error 500.
+    - name: Copy PEM files to Sensu backend
+      copy:
+        src: "files/{{ item }}"
+        dest: "/tmp/{{ item }}"
+        mode: 0744
+      loop:
+        - ca.crt
+        - client.crt
+        - client.key
+
+- name: Converge
+  hosts: all
+  gather_facts: no
+  tasks:
+    - name: Fetch all secrets providers and verify the presence of the
+        default env provider
+      sensu.sensu_go.secrets_provider_info:
+        auth:
+          url: http://localhost:8080
+      register: result
+
+    - ansible.builtin.assert:
+        that:
+          - result.objects | length == 1  # env provider is present by default
+          - result.objects.0.metadata.name == 'env'
+
+    - name: Fail when trying to create a vault provider with missing required params
+      sensu.sensu_go.secrets_provider_vault:
+        auth:
+          url: http://localhost:8080
+        name: my-vault
+      register: result
+      ignore_errors: true
+
+    - ansible.builtin.assert:
+        that:
+          - result is failed
+          - "'state is present but all of the following are missing' in result.msg"
```

Comment:
**No action needed, just a remark.**

I tend to test this using unit tests lately because 1) they are much faster, and 2) I can pin down the behavior more. That second point is important for making sure we do not alter the API in a backward-incompatible way.

**Again, no action needed, just information for the future.**

Reply:
Noted, thank you!